ratpack-manual/src/content/chapters/37-pac4j.md

```diff
@@ -0,0 +1,94 @@
+# Security using pac4j
+
+The `ratpack-pac4j` extension provides authentication support based on [pac4j](https://github.com/pac4j/pac4j).
+
+`pac4j` is a generic authentication engine which is implemented by many frameworks and supports multiple protocols: OAuth, CAS, OpenID (Connect), SAML, Google App Engine and HTTP (form and basic auth).
+
+The `RatpackPac4j` class is the main component to use to manage the security of your web application.
+
+
+## Prerequisite
+
+All authentication mechanisms work by redirecting the user to an identity provider and finishing the login process when he is redirected back to the application with some dedicated credentials.
+To make these authentication processes work and to save the originally requested url, the session is required. That's why the `SessionModule` is necessary:
+
+```language-java
+.registry(Guice.registry(b -> b
```

Comment:
These `language-java` code snippets are tested and require a full Java code listing.

e.g.

import ratpack.guice.Guice;
import ratpack.session.Session;
import ratpack.session.SessionModule;
import ratpack.test.embed.EmbeddedApp;

import static org.junit.Assert.assertEquals;

public class Example {
  public static void main(String[] args) throws Exception {
    EmbeddedApp.of(s -> s
      .registry(
        Guice.registry(b -> b.module(SessionModule.class)))
      .handlers(c ->
        c.all(ctx -> {
          Session session = ctx.get(Session.class);
          ctx.getResponse()
            .send("Context contains session: " + (session != null));
        })))
      .test(testClient ->
        assertEquals("Context contains session: true", testClient.getText())
      );
  }
}